File: go/go-leetcode/examples/median_of_two_sorted_arrays.go
package main

import (
	"fmt"
	"sort"
)

//https://leetcode.com/problems/median-of-two-sorted-arrays/
//https://leetcode.com/submissions/detail/814924203/

func main() {
	fmt.Println(findMedianSortedArrays([]int{1, 2}, []int{3, 4}))
}

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	sortedList := append(nums1, nums2...)
	sort.Ints(sortedList)
	length := len(sortedList)
	if length%2 == 0 {
		indexLower := float64(sortedList[length/2-1])
		indexUpper := float64(sortedList[length/2])
		return (indexLower + indexUpper) / 2
	} else {
		index := float64(length)/2 - .5
		return float64(sortedList[int(index)])
	}
}

File: python/python-leetcode/median_of_two_sorted_arrays.py
# https://leetcode.com/problems/median-of-two-sorted-arrays/
# https://leetcode.com/submissions/detail/814227911/
from typing import List


class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        listed = sorted(nums1 + nums2)
        listed_length = len(listed)
        if listed_length % 2 == 0:
            index_lower = listed[int(listed_length / 2 - 1)]
            index_upper = listed[int(listed_length / 2)]
            return (index_upper + index_lower) / 2
        else:
            return listed[int(listed_length / 2 - 0.5)]


sol = Solution()
print(sol.findMedianSortedArrays([1, 1], [1, 2]))
